src/expression/expression.test.ts

```diff
@@ -629,3 +629,43 @@ describe('slice expression', () => {
         expect((response.value as StyleExpression)?.evaluate({zoom: 20})).toEqual([]);
     });
 });
+
+describe('projection expression', () => {
+
+    test('step', () => {
+        const response = createExpression(['step', ['zoom'], 'vertical-perspective', 10, 'mercator']);
+
+        if (response.result === 'success') {
+            expect(response.value.evaluate({zoom: 5})).toBe('vertical-perspective');
+            expect(response.value.evaluate({zoom: 10})).toBe('mercator');
+            expect(response.value.evaluate({zoom: 11})).toBe('mercator');
+        } else {
+            throw new Error('Failed to parse Step expression');
+        }
+    })
+
+    test('step array', () => {
+        const response = createExpression(['step', ['zoom'], ['literal', ['vertical-perspective', 'mercator', 0.5]], 10, 'mercator']);
+
+        if (response.result === 'success') {
+            expect(response.value.evaluate({zoom: 5})).toStrictEqual(['vertical-perspective', 'mercator', 0.5]);
+            expect(response.value.evaluate({zoom: 10})).toBe('mercator');
+            expect(response.value.evaluate({zoom: 11})).toBe('mercator');
+        } else {
+            throw new Error('Failed to parse Step expression');
+        }
+    })
+
+    test('interpolate', () => {
+        const response = createExpression(['interpolate', ['linear'], ['zoom'], 8, 'vertical-perspective', 10, 'mercator']);
```

Comment:
since we're working with strings as interpolate values, it's quite tricky to have a constraint that doesn't allow strings as interpolate values. The "vertical-perspective" will be parsed into `kind`: 'string'` rather than kind: projectionDefinition, and I don't know how I can pass the projectionDefinition information

Reply:
This is not different than color parsing from string. The kind should be coming from the spec definition.

Reply:
I haven't see color parsing in these tests, I'm guessing it is related to the fact the `kind` need to be set and this is done in the integration tests...

Reply:
It's true that color has a way to not the expression engine that this part should be cast to type color.
But when using in the style you can simply use "red", "green" etc, this is similar to `mercator`.
We could in theory in the future add a `proj` operator like rgba, but I don't think we need it now.
The epxression engine knows the expected type based on the kind from the spec.
If you would try to do the same test with `createExpression(['interpolate', ['linear'], ['zoom'], 0, 'red', 10, 
'green']);` I believe it will fail like it would for expression.

Reply:
I've fixed the tests by adding the relevant property spec parameter to help the expression engine understand that projection definition is expected.